lib/matplotlib/axes/_axes.py

```diff
@@ -3363,14 +3354,18 @@ def _upcast_err(err):
                     # fails.
                     isinstance(cbook.safe_first_element(err), np.ndarray)
             ):
-                # grab the type of the first element, we will try to promote
-                # the outer container to match the inner container
+                # Get the type of the first element
                 atype = type(cbook.safe_first_element(err))
-                # you can not directly pass data to the init of `np.ndarray`
+                # If atype is np.ndarray, promote the outer container to
```

Comment:
In all cases we want to promote the outer to match the type of the inner.  The problem we are trying to solve is to turn `[TYPE([1, 2]), TYPE([3, 4]]` -> `TYPE([[1, 2], [3, 4]])`  where TYPE may be a ndarray subclass or something more exotic.  

If we were to just do `np.asarray(...)` we would drop any type information that the containers were carrying,  just get naive floats, and lose the information about what TYPE was or any extra metadata it was carrying.

If we do `TYPE([...])` then it fails for numpy due to ndarray not having a `__init__` signature that works with that!